.circleci/config.yml

```diff
@@ -0,0 +1,40 @@
+####################
+# CircleCI configuration reference:
+#   https://circleci.com/docs/2.0/configuration-reference
+####################
+# CircleCI built-in environment variables:
+#   https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables
+####################
+
+version: 2
+jobs:
+  build:
+    docker:
+      - image: mozilla/sbt:8u171_0.13.13
+    steps:
+      - checkout
+      - run:
+          name: Test
+          command: |
+            sbt coverage scalastyle test:scalastyle test coverageReport
+      - run:
+          name: Submit coverage data
+          command: |
+            bash <(curl -s https://codecov.io/bash)
+      - run:
+          name: Early return if this build is from a forked PR
+          command: |
+            if [ -n "$CIRCLE_PR_NUMBER" ] || [ "$CIRCLE_BRANCH" != "master" ]; then
+              echo "Stopping here; we only publish for pushes to the master branch"
+              circleci step halt
+            fi
+      - run:
+          name: Publish
+          command: |
+            sbt publish spPublish
```

Comment:
Does this effectively publish on every commit? Is that actually what we want? Can we make it only publish on a tag?

Reply:
The previous command does an early return if this is not a commit to master, but it's a bit ugly.

A less efficient but more clear way of doing this would be to split out publishing to a separate "job" and then have the workflow specify `branch: master` for whether to execute that job. I think I'll go ahead and make that change.